pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesources
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/migration"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+)
+
+// NodeResourcesLeastAllocated is a score plugin that favors nodes with fewer allocation requested resources based on requested resources.
+type NodeResourcesLeastAllocated struct {
+	handle framework.FrameworkHandle
+}
+
+var _ = framework.ScorePlugin(&NodeResourcesLeastAllocated{})
+
+// LeastAllocatedName is the name of the plugin used in the plugin registry and configurations.
+const LeastAllocatedName = "NodeResourcesLeastAllocated"
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (nrla *NodeResourcesLeastAllocated) Name() string {
```

Comment:
```suggestion
func (pl *NodeResourcesLeastAllocated) Name() string {
```

Same as other plugins